src/Pos/Util/UserSecret.hs

```diff
@@ -1,8 +1,13 @@
+{-# LANGUAGE CPP             #-}
 {-# LANGUAGE TemplateHaskell #-}
 
 -- | Secret key file storage and management functions based on file
 -- locking.
 
+#if !defined(mingw32_HOST_OS) && !defined(__MINGW32__)
```

Comment:
It turned out that just `#ifndef mingw32_HOST_OS` is enough: https://ghc.haskell.org/trac/ghc/ticket/9945#comment:15